src/lib/Gateway/UpdateSerializerFactory.php

```
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Solr\Gateway;
+
+use eZ\Publish\SPI\Exception\InvalidArgumentException;
+
+class UpdateSerializerFactory
+{
+    /** @var \Ibexa\Solr\Gateway\UpdateSerializerInterface[]|iterable */
+    private $serializers;
+
+    /**
+     * @param array<\Ibexa\Solr\Gateway\UpdateSerializerInterface> $serializers
+     */
+    public function __construct(iterable $serializers)
```

This may confuse the static analysis and hide an error when someone tries to access this like an array, when it may not be one.

Side note:
Additionally, you can make use of `index_by` functionality of `!tagged_iterator` to remove the need for serializers to have `getSupportedFormat()` method. This would also make it possible for a serializer to be registered more than once, for multiple formats.

---

> This may confuse the static analysis and hide an error when someone tries to access this like an array, when it may not be one.

So what should be done here instead?

> 
> Side note: Additionally, you can make use of `index_by` functionality of `!tagged_iterator` to remove the need for serializers to have `getSupportedFormat()` method. This would also make it possible for a serializer to be registered more than once, for multiple formats.

This is done on purpose, it's not used just by the factory:
>  `'Content-Type' => 'application/' . $this->updateSerializer->getSupportedFormat(),`

Serializer's responsibility is supporting one format compatible with Solr Content-Type. It's not within the scope of DI.

---

> > This may confuse the static analysis and hide an error when someone tries to access this like an array, when it may not be one.
> 
> So what should be done here instead?
> 

Fixed via bcff6e7